src/processor/result/factorized_table.cpp

```diff
@@ -284,9 +284,10 @@ void FactorizedTable::copyToInMemList(ft_col_idx_t colIdx, vector<ft_tuple_idx_t
     NodeIDCompressionScheme* nodeIDCompressionScheme) const {
     auto column = tableSchema->getColumn(colIdx);
     assert(column->isFlat() == true);
-    auto numBytesPerValue = nodeIDCompressionScheme == nullptr ?
-                                Types::getDataTypeSize(type) :
-                                nodeIDCompressionScheme->getNumBytesForNodeIDAfterCompression();
+    auto numBytesPerValue =
+        nodeIDCompressionScheme == nullptr ?
+            (type.typeID == INTERNAL_ID ? sizeof(offset_t) : Types::getDataTypeSize(type)) :
```

Comment:
why do u need to check `(type.typeID == INTERNAL_ID ? sizeof(offset_t) : Types::getDataTypeSize(type))` we don't seem to need this check before

Reply:
Because we do a relID optimization here, we only store the relOffset in the storage layer.